numpy/_core/src/umath/log1p_complex.h

```diff
@@ -0,0 +1,260 @@
+#ifndef LOG1P_COMPLEX_H
+#define LOG1P_COMPLEX_H
+
+#include <Python.h>
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_math.h"
+
+#include <cmath>
+#include <complex>
+#include <limits>
+
+// For memcpy
+#include <cstring>
+
+
+//
+// Trivial C++ wrappers for several npy_* functions.
+//
+
+#define CPP_WRAP1(name) \
+inline float name(const float x)                \
+{                                               \
+    return ::npy_ ## name ## f(x);              \
+}                                               \
+inline double name(const double x)              \
+{                                               \
+    return ::npy_ ## name(x);                   \
+}                                               \
+inline long double name(const long double x)    \
+{                                               \
+    return ::npy_ ## name ## l(x);              \
+}                                               \
+
+#define CPP_WRAP2(name) \
+inline float name(const float x,                \
+                  const float y)                \
+{                                               \
+    return ::npy_ ## name ## f(x, y);           \
+}                                               \
+inline double name(const double x,              \
+                   const double y)              \
+{                                               \
+    return ::npy_ ## name(x, y);                \
+}                                               \
+inline long double name(const long double x,    \
+                        const long double y)    \
+{                                               \
+    return ::npy_ ## name ## l(x, y);           \
+}                                               \
+
+
+namespace npy {
+
+CPP_WRAP1(fabs)
+CPP_WRAP1(log)
+CPP_WRAP1(log1p)
+CPP_WRAP2(atan2)
+CPP_WRAP2(hypot)
+
+}
```

Comment:
I guess we don't have a nicer C++ pattern/common spot for these yet, so this looks fine.